### PR TITLE
Bug fix 189 balance

### PR DESCRIPTION
**Pull request by CFDevelopment**
- Balance is loaded async 
- Ether - ERC20 abstracted to scripts directory for modularity
- Balance Reducer created to track tokens and balances

### CODE REVIEW

File: src/reducers/wallet/Balance.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    case FETCHING_ETH_PRICE_DATA_FAIL: 
+      return { ...state, isFetching: false, hasError: true, errorMessage: action.err };   
+    case SET_WALLET_TOKENS_BALANCES:   
+      return { ...state, walletTokens: action.payload };
+    case CALCULATE_WALLET_BALANCE:
+      const key = Object.keys(state.walletTokens[0]); 
+      let quant = state.walletTokens[0][key[1]];    
+      let usdPrice = state.tokenConversions["ETH"].USD;     
+      let total = quant*usdPrice;
+      return { ...state, initialBalance: total };
```
Comment (GregTheGreek) [inline src/reducers/wallet/Balance.js]: This logic should happen outside of the reducer. Move it to the appropriate action

File: src/scripts/tokenBalances.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * Any subsequent coins will have an index of 1 or greater, unless they have removed ether from their wallet.
+ * This use case will be address in a subsequent branch.
+ */
+
+import provider from '../constants/Providers';
+import ERC20ABI from '../constants/data/json/ERC20ABI.json';
+
+const ethers = require('ethers');
+
+const processAllTokenBalances = async (privateKey, dataSet) => {
```
Comment (GregTheGreek) [inline src/scripts/tokenBalances.js]: this function has a lot of console.log()

File: src/scripts/tokenBalances.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        ethObj.amount = parsedEtherBalance;        
+        tokenSymbolString += `${key[0]}`;   
+        if (dataSet.length > 1) tokenSymbolString += ',';
+        tokenHoldings.push(ethObj);
+        break;
+      default: 
+        key = Object.keys(dataSet[i]);    
+        const contract = new ethers.Contract(key[1], ERC20ABI, provider); //key[1] is contract address
+        const tokenBalance = await contract.balanceOf(wallet.address); 
+        key = Object.keys(dataSet[i]);
```
Comment (GregTheGreek) [inline src/scripts/tokenBalances.js]: this is also on line 39

File: src/scripts/tokenBalances.js (hunk 2 above)
Comment (CFDevelopment) [reply src/scripts/tokenBalances.js]: @GregTheGreek fixed

File: src/scripts/tokenBalances.js (hunk 3 above)
Comment (CFDevelopment) [reply src/scripts/tokenBalances.js]: fixed

File: src/reducers/wallet/Balance.js (hunk 1 above)
Comment (CFDevelopment) [reply src/reducers/wallet/Balance.js]: @GregTheGreek done

File: src/scripts/tokenBalances.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  const parsedEtherBalance = String(ethers.utils.formatEther(balance));
+  console.log('Ethereum Balance', parsedEtherBalance);
+  return parsedEtherBalance;
+};
+
+getERC20Balance = async (contractAdd) => {
+  const contract = new ethers.Contract(contractAdd, ERC20ABI, provider);
+  const tokenBalance = await contract.balanceOf(wallet.address);
+  const parsedTokenBalance = String(tokenBalance);
+  console.log('Token Balance', parsedTokenBalance);
```
Comment (GregTheGreek) [inline src/scripts/tokenBalances.js]: Log

File: src/screens/main/portfolio/index.js [hunk 5]
```diff
@@ -167,19 +119,16 @@ class Portfolio extends Component {
   }
 
   handleListRefresh = () => {
-    this.setState({ refresh: true, currencyIndex: 0 },
-      () => {
-        this.tokenBalanceLoop();
-      });
+    console.log('List refreshed');
```
Comment (GregTheGreek) [inline src/screens/main/portfolio/index.js]: Log

File: src/scripts/tokenBalances.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  let tokenApiRequestString = '';
+  wallet = new ethers.Wallet(privateKey);
+  wallet.provider = provider;
+  console.log(privateKey, dataSet, dataSet.length);
+  for (let i = 0; i < dataSet.length; i++) {
+    let tokenObj = {};
+    tokenObj.symbol = dataSet[i].symbol;
+    if (dataSet[i].contractAddress === '') {
+      await this.getEthereumBalance(wallet.address)
+        .then((response) => {
```
Comment (GregTheGreek) [inline src/scripts/tokenBalances.js]: Log

File: src/screens/main/portfolio/index.js (hunk 5 above)
Comment (CFDevelopment) [reply src/screens/main/portfolio/index.js]: @GregTheGreek done

File: src/scripts/tokenBalances.js (hunk 6 above)
Comment (CFDevelopment) [reply src/scripts/tokenBalances.js]: @GregTheGreek done

File: src/scripts/tokenBalances.js (hunk 4 above)
Comment (CFDevelopment) [reply src/scripts/tokenBalances.js]: done

**Pull request merged**